src/main/java/seedu/address/logic/commands/schedule/FindCommonTimingCommand.java

```diff
@@ -55,51 +58,38 @@ public FindCommonTimingCommand(IsTagInPersonPredicate predicate, LocalDate date)
      * @param event event that needs to be accounted for
      */
     public void blockTimeSlots(int[] timeSlots, Event event) {
-        LocalTime startTime = event.getTime();
+        Event eventAtCurrentDate = event.getEventAtDate(date);
+        LocalTime startTime = eventAtCurrentDate.getTime();
         int hours = startTime.getHour();
         int minutes = startTime.getMinute();
         int startSlot = hours * 2;
         if (minutes >= 30) {
             startSlot += 1;
         }
-
-        LocalTime endTime = event.getEndTime();
+        LocalTime endTime = eventAtCurrentDate.getEndTime();
         int endHours = endTime.getHour();
         int endMinutes = endTime.getMinute();
         int endSlot = endHours * 2;
         if (endMinutes > 30) {
+            endSlot += 2;
+        } else if (endMinutes > 0) {
             endSlot += 1;
         }
 
-        for (int i = startSlot; i <= endSlot; i++) {
+        Duration duration = eventAtCurrentDate.getDuration();
+        if (duration == Duration.ofDays(1)) {
             //set as busy
-            timeSlots[i] = 1;
+            Arrays.fill(timeSlots, 1);
```

Comment:
This part is not needed since we already set endSlot to 48 when endTime is 00:00.
If the startTime is also 00:00, then startSlot will be 0. We will be able to cover from 0 to 48 like this. There is no need to consider this special case.

Reply:
Alright